### PR TITLE
SHARE-150 Redesign of followers

### PR DESCRIPTION
**Pull request by arminpasalic10**
- The followers are moved from the profile page into its own category, which can be accessed from the side panel
- The new followers page consists of two sections: following & followers
- Users can follow/unfollow other users directly from this page by clicking the button on the right side
- Changed the look of the follow button when visiting the profile of another user.
- Modified the design of "follow notifications"
- Behat test

---
### Your checklist for this pull request
Please review the [contributing guidelines](https://github.com/Catrobat/Catroweb-Symfony/blob/develop/.github/contributing.md) and [wiki pages](https://github.com/Catrobat/catroweb-Symfony/wiki/) of this repository.

- [X] Include the name and id of the Jira ticket in the PR’s title eg.: `SHARE-666 The devils ticket`
- [X] Choose the proper base branch (*develop*)
- [X] Confirm that the changes follow the project’s coding guidelines
- [X] Verify that the changes generate no warnings and errors 
- [X] Verify to commit no other files than the intentionally changed ones
- [X] Include reasonable and readable tests verifying the added or changed behavior
- [X] Verify that all tests are passing, if not please state the test cases in the [section](#Tests) below
- [X] Perform a self-review of the changes
- [X] Stick to the project’s git workflow (rebase and squash your commits)
- [X] Verify that your changes do not have any conflicts with the base branch
- [X] Put your ticket into the `Code Review` section in [Jira](https://jira.catrob.at/)
- [X] Post a message in the *#catroweb* [Slack channel](https://catrobat.slack.com) and ask for a code reviewer

### Additional Description
`TODO: Add additional information that is not in your commit-message here`

In the task description, it is requested that the design of follower notifications should be changed and that every follower notification should look the same as the blocks in the "Followers & Following" section (avatar on the left, other information on the right). However, by doing that we would get a design, where the follower notifications (and only those notifications) would have a completely different structure from all other notification types. After a discussion with one of the seniors, we agreed that the design should stay consistent with other notification types, so only small modifications were made for follower notifications (a follow button was added). If the design team insists that the structure of follower notfications has to be identical to the block structure that was described in the ticket description, you can simply switch the design by uncommenting the code in "follow_notification.html.twig".

### Tests - additional information
`TODO: add additional information about testruns here`

### CODE REVIEW

**Review by dmetzner — CHANGES_REQUESTED** [review]
* Please check/fix your code style. Just use Phpstorm to reformat your changed files. Or you can check it manually with the check style command.
    E.g. 
    * consistent indentations
    * `{` on a new line
    * many unnecessary new lines,
    * sometimes a missing new line
    * Please always use braces for if conditions

* Please remove all the unused code you just commented out.

* Consider the usage of default parameters for followers.js. Would be cleaner than always manually setting "5,5,5,5,5,5," 

* Followers category should only be visible when logged in the navbar

* If the follow categories are empty, add a text telling the user he has no followers, or isn't following anybody.

* Following is not that dramatic that we should need an "Are you sure .." pop up. Just do it. Social media platforms like Instagram do the same. Only unfollowing should use a pop up.
Btw the cancel button is not working. (Uncaught in promise cancel)

* Remove follower logic from the profile loader since there should not be any follower stuff anymore. **Edit: might can get rid of the file completely**

* The notifications are kinda buggy. The same user can trigger an unlimited number of follow notifications. Do not create a new follow notification if the user already has the same notification. 

* Looking at the UX designs the navbar button should only contain "Follower"

* Looking at the UX design there should only be one entry at follower entry at once, to show more the category has to be extended using a button (use lazy loading)

* Shouldn't other notifications like; reactions, comments now contain a follow/unfollow button?

* mobile is everything, have you checked your design for small screens? it looks kinda buggy to me.

image

image

image



* Tests would have passed

**Review by dmetzner — CHANGES_REQUESTED** [review]
as discussed in discord

File: translations/catroweb.en.yml [hunk 1]
```diff
@@ -848,6 +853,7 @@ catro-notifications:
   follow:
     title: 'New Follower!'
     message: 'User %user_link% follows you now.'
+    message2: '%user_link%'
```
Comment (dmetzner) [inline translations/catroweb.en.yml]: why would you need something like this in the translations?
No one will ever be able to translate it

**Comment by HCrane** [comment]
Current code will be merged via other PR. Closing this as it is not neccesary to have 2 PRs with the same code.

**Pull request closed without merging**